### PR TITLE
adjust DataOffset so SocketAddress.ToString() works same way as Windows 

### PR DESCRIPTION
**Pull request by wfurt**
I tested this change on osx 10.13, Linux, Windows and Linux-arm.

### CODE REVIEW

**Comment by wfurt** [comment]
@dotnet-bot test Outerloop Windows x64 Debug Build
@dotnet-bot test Outerloop Linux x64 Debug Build
@dotnet-bot test Outerloop Linux arm Debug Build
@dotnet-bot test Outerloop NETFX x86 Debug Build
@dotnet-bot test Outerloop UWP CoreCLR x64 Debug Build
@dotnet-bot test Outerloop OSX x64 Debug Build

File: src/System.Net.Primitives/tests/PalTests/SocketAddressPalTest.cs [hunk 1]
```diff
@@ -188,5 +188,15 @@ public void IPv6AddressAndPort_Get_Set(byte[] address, uint scope, ushort port)
             Assert.Equal(scope, actualScope);
             Assert.Equal(port, SocketAddressPal.GetPort(buffer));
         }
+
+        // This is platform specific test. So far Windows, Linux and OSX use same layout.
```
Comment (davidsh) [inline src/System.Net.Primitives/tests/PalTests/SocketAddressPalTest.cs]: nit: This comment is confusing.  This test is not "platform-specific".  It is defined as a "Fact" and will run for any platform.  I would delete this comment.

File: src/System.Net.Primitives/tests/PalTests/SocketAddressPalTest.cs [hunk 2]
```diff
@@ -188,5 +188,15 @@ public void IPv6AddressAndPort_Get_Set(byte[] address, uint scope, ushort port)
...
+        [Fact]
+        public void SocketAddress_To_String()
```
Comment (davidsh) [inline src/System.Net.Primitives/tests/PalTests/SocketAddressPalTest.cs]: nit: This test method name doesn't follow our convention. The first part of the name is the property/method being tested, etc.  I would suggest this be "ToString_ValidSocketAddress_ExpectedFormat()

File: src/Common/src/System/Net/SocketAddressPal.Unix.cs [hunk 3]
```diff
@@ -12,7 +12,7 @@ namespace System.Net
 {
     internal static class SocketAddressPal
     {
-        public const int DataOffset = 0;
+        public const int DataOffset = 2;
```
[the remark this answers is not in the document]
Comment (wfurt) [reply src/Common/src/System/Net/SocketAddressPal.Unix.cs]: I looked at AIX and FreeBSD and the offset would also be 2. So it seems like it is very unlikely we will ever see different value. (at least in near future)

[the remark this answers is not in the document]
Comment (davidsh) [reply src/Common/src/System/Net/SocketAddressPal.Unix.cs]: >then it probably doesn't need to be in the PAL
>---So it seems like it is very unlikely we will ever see different value. (at least in near future)

Does that mean that you are revising the PR to remove the `DataOffset` value and move it away from the SocketAddressPal class?

[the remark this answers is not in the document]
Comment (wfurt) [reply src/Common/src/System/Net/SocketAddressPal.Unix.cs]: yes. it is now private field in SocketAddress it self.

**Review by davidsh — APPROVED** [review]
You should re-run all Outerloop tests before merge.

**Comment by wfurt** [comment]
@dotnet-bot test Outerloop Windows x64 Debug Build
@dotnet-bot test Outerloop Linux x64 Debug Build
@dotnet-bot test Outerloop Linux arm Debug Build
@dotnet-bot test Outerloop NETFX x86 Debug Build
@dotnet-bot test Outerloop UWP CoreCLR x64 Debug Build
@dotnet-bot test Outerloop OSX x64 Debug Build

**Comment by wfurt** [comment]
Test failures unrelated. fixes #30523

**Pull request merged**